src/Features/Core/Portable/AddRequiredParentheses/AbstractAddRequiredParenthesesDiagnosticAnalyzer.cs

```diff
@@ -139,8 +139,9 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
                 var properties = GetProperties(includeInFixAll, equivalenceKey);
 
                 context.ReportDiagnostic(Diagnostic.Create(
-                    GetDescriptorWithSeverity(severity),
+                    Descriptor,
                     operatorToken.GetLocation(),
+                    severity,
```

Comment:
>severity [](start = 20, length = 8)

Unrelated to your PR, but I wonder what would be the effective severity of the diagnostic here if there is a conflicting ruleset entry and tools/option setting for this rule ID, we seem to be only using the later for computing effective severity.

I really wish our IDE analyzers can get out of this business of computing effective severity based on options. This is something that should be dealt with by common infrastructure, just like ruleset is converted into compilation options.

Reply:
Ah, I presume it will work fine because compilation options will override the severity reported by the analyzer. Still will be good to clean this up.

---
In reply to: [192830806](https://github.com/dotnet/roslyn/pull/27220#discussion_r192830806) [](ancestors = 192830806)

Reply:
> I really wish our IDE analyzers can get out of this business of computing effective severity based on options.

This is going to be especially difficult because some analyzers report the same diagnostic with different severities. It would be good to have some sort of solution though. Perhaps a suffix that can be included with the diagnostic descriptor and included in .editorconfig (and here I'm thinking of the new [analyzer configuration support](https://github.com/agocke/roslyn/blob/5512ea8fd8d30b9b785cdce70d0ed702bf231e96/.editorconfig#L8)), and analyzers select the appropriate reporting configuration only indirectly by ensuring the correct diagnostic descriptor is used when reporting the diagnostic. Then at the time of presentation, we drop the suffix so it looks like the rule was simply configured to report at different severities depending on context.